vyper/semantics/validation/utils.py

```diff
@@ -204,6 +204,18 @@ def types_from_Compare(self, node):
         return [BoolDefinition()]
 
     def types_from_Call(self, node):
+        # workaround for `_abi_decode` to derive type from LHS
+        if isinstance(node.func, vy_ast.Name) and node.func.id == "_abi_decode":
+            parent_node = node.get_ancestor()
+            if isinstance(parent_node, (vy_ast.Assign, vy_ast.AnnAssign)):
+                return [parent_node._metadata["type"]]
+
+            elif isinstance(parent_node, vy_ast.Return):
+                func_node = node.get_ancestor(vy_ast.FunctionDef)
+                return [func_node._metadata["type"].return_type]
+
+            raise StructureException("Unable to determine the return type of abi_decode", node)
```

Comment:
it's good that this logic has been moved to the type checker, but this doesn't handle cases where the `_abi_decode()` call is not in a return or assignment position, ex. in some expression position (like `max(0, _abi_decode(...))`).

i wonder if there is a slicker way to signal that the typecheck should always pass? maybe a special case in `validate_expected_type()`?